File: 6.C# Fundamentals/OOP Basics/2.Defining_Classes_Exce/DefiningClassesExce/4.Opinion_Poll/Person.cs
class Person
{

    public string Name { get; set; }
    public int Age { get; set; }

    /*Zd 2*/

    public Person()
    {
        this.Name = "No name";
        this.Age = 1;
    }


    public Person(int age)
    {
        this.Name = "No name";
        this.Age = age;
    }

    public Person(string name, int age)
    {
        this.Name = name;
        this.Age = age;
    }
}

File: 7.JS Core/JS Advanced/19.Modules/Inheritance/Person.js


let Entity = require('./Entity');
let Dog = require('./Dog');


class Person extends Entity
{
    constructor(name, phrase, dog) 
    {
        super(name);

        this.phrase = phrase;
        this.dog = dog;
    }

    saySomething(){
        return `${this.name} says: ${this.phrase}${this.dog.name} barks!`;
    }
}

/*
let d = new Dog('Gafi');
let person = new Person("Nasko", 'My dog', d);

console.log(person.saySomething())
*/

module.exports = Person;





